docs/token-builder.md

```diff
@@ -0,0 +1,288 @@
+# Token Builder API
+
+## Overview
+
+The Token Builder is comprised of two parts: the `RequestBuilder` class and a variety of `Token` type classes.
+
+Each of the token classes has its own set of validations which are run when the token is instantiated. Any errors are added to an `errors` property on that instance of the token class.
+
+In `sorcerer`, the Token Parser will extract the tokens from request body and replace any occurrence of the token in an API data source's `url`, `body`, or `headers` with the token's computed value.
+
+## Token Types
+
+The Token Builder API includes several token utility classes, each serving a unique purpose.
+
+Currently supported tokens are:
+
+- ReplaceToken
+- ReplaceLargeToken
+- SecretToken
+- HmacToken
+- Sha1Token
+
+### ReplaceToken
+
+Replaces token with the `value` included in the token options. The length of this value must be less than 100 characters.
+
+Example:
+
+```jsx
+const options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
```

Comment:
suggestion: When it comes to documentation, IMO as a best practice, we should start off with the "simplest" examples. In this case, we should strip out `cacheOverride` and `skipCache`. We should let them assume the defaults of what the classes were meant to do, and provide more advanced examples further down in their respective explanations for `cacheOverride` and `skipCache`.

The downside in providing them initially is that users think that this is the preferred usage. We should encourage the default usage and only provide how to non-default options as exceptions (with guidance on how the exception works).